src/server/mod.rs

```diff
@@ -98,10 +98,16 @@ impl BlockingRequestAction for InitializeRequest {
         let init_options: InitializationOptions = params
             .initialization_options
             .as_ref()
-            .and_then(|options| serde_json::from_value(options.to_owned()).ok())
+            .and_then(|options| {
+                let de = serde_json::from_value(options.to_owned());
```

Comment:
Since we accept `params` by-move, we could probably just `map` and move the inner `options` instead of cloning via `to_owned` here

Reply:
'params' is borrowed later, presumably the reason it was cloned. You can `.take()` the options, but perhaps this could cause future confusion.

As it's debatable what's better I just left it as is.

Reply:
Ah, you're right! Didn't notice that.